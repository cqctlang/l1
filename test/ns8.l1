// circular definitions should be caught
ns = @names c32le {
	struct X {
		@0x0 X x;
		@0x4;
	};
	typedef X X;
	@0x0 struct X x;
};
dom = mkdom(ns, as);
printf("%t\n", &dom`x->x);

