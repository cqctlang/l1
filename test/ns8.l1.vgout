entry: var params, 12 locals, 0 temps
error: circular definition in name space: typedef X
             mknsraw	(builtin function)
               entry	(<stdin>:0)
