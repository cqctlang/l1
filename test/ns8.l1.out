error: circular definition in name space: typedef X
	#0                 mknsraw	(builtin function)
	#1                   entry	(ns8.l1:9)
